src/ast/passes/codegen_llvm.h

```diff
@@ -248,6 +249,13 @@ class CodegenLLVM : public Visitor {
 
   Value *createFmtString(int print_id);
 
+  Value *createTupleAllocation(const SizedType &tuple_type,
+                               const std::string &name,
+                               const location &loc);
+  Value *createGetFmtStringArgsAllocation(StructType *struct_type,
+                                          const std::string &name,
+                                          const location &loc);
+
```

Comment:
Wondering if there's a way to prevent proliferation of wrappers like this. Seems like it would be good to keep the "stack or offstack" decision in a central place so new offstack types can't easily forget to check on stack_threshold.

I can't think of any better ideas at the moment so just consider this a non-blocking idea to ponder.

Reply:
Yea the problem is we depend on iterating `async_ids` if using scratch, else not iterating it. So its hard to move into irbuilderbpf.cpp. I guess we could do this with a callback...let me experiment with that approach.

Reply:
Addressed in latest version using callbacks